ports/servo/browser.rs

```diff
@@ -349,20 +363,15 @@ fn platform_get_selected_devices(devices: Vec<String>, sender: IpcSender<Option<
 }
 
 #[cfg(any(target_os = "macos", target_os = "linux", target_os = "windows"))]
-fn platform_get_selected_files(patterns: Vec<FilterPattern>,
+fn platform_get_selected_files(filters: Vec<String>,
                                multiple_files: bool,
                                sender: IpcSender<Option<Vec<String>>>) {
     let picker_name = if multiple_files { "Pick files" } else { "Pick a file" };
 
     thread::Builder::new().name(picker_name.to_owned()).spawn(move || {
-        let mut filter = vec![];
-        for p in patterns {
-            let s = "*.".to_string() + &p.0;
-            filter.push(s)
-        }
 
-        let filter_ref = &(filter.iter().map(|s| s.as_str()).collect::<Vec<&str>>()[..]);
-        let filter_opt = if filter.len() > 0 { Some((filter_ref, "")) } else { None };
+        let filter_ref = &(filters.iter().map(|s| s.as_str()).collect::<Vec<&str>>()[..]);
+        let filter_opt = if filters.len() > 0 { Some((filter_ref, "")) } else { None };
```

Comment:
Could we use `FilterPattern` here if we move its definition to the embedder_traits? It's a bit annoying that we have to move that logic (FilterPattern to string) in `filemanager_thread.rs`. Feels like the embedder code is a better place. I'm not sure what's the best approach here.